src/Client/ConfigWindow.java

```diff
@@ -4059,31 +4434,87 @@ public void actionPerformed(ActionEvent e) {
-    replayPanelReplayFolderBasePathTextFieldLabel.setAlignmentY(0.9f);
-
-    replayPanelReplayFolderBasePathTextField = new JTextField();
-    replayPanelReplayFolderBasePathTextFieldPanel.add(replayPanelReplayFolderBasePathTextField);
-    replayPanelReplayFolderBasePathTextField.setMinimumSize(osScaleMul(new Dimension(100, 28)));
-    replayPanelReplayFolderBasePathTextField.setMaximumSize(
+    JLabel replayPanelReplayBasePathTextFieldLabel = new JLabel("Replay Folder Location: ");
+    replayPanelReplayBasePathTextFieldLabel.setToolTipText(
+        "Default location for the replay queue file chooser");
+    replayPanelReplayBasePathTextFieldPanel.add(replayPanelReplayBasePathTextFieldLabel);
+    replayPanelReplayBasePathTextFieldLabel.setAlignmentY(replayLabelAlignment);
+
+    replayPanelReplayBasePathTextField = new JTextField();
+    replayPanelReplayBasePathTextFieldPanel.add(replayPanelReplayBasePathTextField);
+    replayPanelReplayBasePathTextField.setMinimumSize(osScaleMul(new Dimension(100, 28)));
+    replayPanelReplayBasePathTextField.setMaximumSize(
         new Dimension(Short.MAX_VALUE, osScaleMul(28)));
-    replayPanelReplayFolderBasePathTextField.setAlignmentY(0.75f);
+    replayPanelReplayBasePathTextField.setAlignmentY(0.75f);
+    replayPanelReplayBasePathTextField.setToolTipText(
+        "Any string of characters you enter into this field will be removed from the Folder Path column in the Replay Queue window.");
+
+    if (Util.isUsingFlatLAFTheme()) {
+      replayPanelReplayBasePathTextFieldPanel.add(
+          Box.createRigidArea(osScaleMul(new Dimension(6, 0))));
+    }
+
+    File replayBaseDir = new File(Settings.REPLAY_BASE_PATH.get("custom"));
+    JFileChooser replayBaseDirChooser = new JFileChooser(replayBaseDir);
+    replayBaseDirChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+
+    JButton replayPanelReplayBasePathSetDirButton =
+        addButton("Set", replayPanelReplayBasePathTextFieldPanel, Component.RIGHT_ALIGNMENT);
+    replayPanelReplayBasePathSetDirButton.setAlignmentY(replayButtonAlignment);
+    replayPanelReplayBasePathSetDirButton.addActionListener(
+        new ActionListener() {
+          @Override
+          public void actionPerformed(ActionEvent e) {
+            int choice = replayBaseDirChooser.showOpenDialog(Launcher.getConfigWindow().frame);
+            if (choice == JFileChooser.APPROVE_OPTION) {
+              File replayBaseDir = replayBaseDirChooser.getSelectedFile();
+              if (validateChosenDirectory(replayBaseDir)) {
+                reindexSearch(
+                    () -> {
+                      Settings.REPLAY_BASE_PATH.put("custom", replayBaseDir.toString());
+                      replayBaseDirChooser.setCurrentDirectory(replayBaseDir);
+                      Settings.save();
+                      synchronizeGuiValues();
+                    });
+              }
+            }
+          }
+        });
```

Comment:
Gotcha, I wasn't super familiar with the replay queue workflow until you pointed it out here, thank you.

Cleaned up the logic, it will always add trailing slashes when they're missing now, for all related settings.

Tested on linux, win, winxp, and macos.